### PR TITLE
Production planning DTOs

### PR DESCRIPTION
**Pull request by vitlinda**
This pr introduces the `Production Planning` and the `Product` DTOs.

### CODE REVIEW

File: production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import dev.atedeg.mdm.productionplanning.OutgoingEvent.*
+import dev.atedeg.mdm.productionplanning.dto.OrderIDDTO.given
+import dev.atedeg.mdm.productionplanning.dto.QuantityDTO.given
+import dev.atedeg.mdm.products.dto.ProductDTO
+import dev.atedeg.mdm.products.dto.ProductDTO.given
+import dev.atedeg.mdm.utils.serialization.DTO
+import dev.atedeg.mdm.utils.serialization.DTOGenerators.*
+import dev.atedeg.mdm.utils.serialization.DTOOps.*
+
+// incoming events
```
Comment (nicolasfara) [inline production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala]: remove the comments

File: production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+final case class NewOrderReceivedDTO(order: OrderDTO)
+final case class OrderDTO(orderID: String, requiredBy: String, orderedProducts: List[OrderedProductDTO])
+final case class OrderedProductDTO(product: ProductDTO, quantity: Int)
+
+object NewOrderReceivedDTO:
+  given DTO[NewOrderReceived, NewOrderReceivedDTO] = interCaseClassDTO
+  private given DTO[Order, OrderDTO] = interCaseClassDTO
+  private given DTO[OrderedProduct, OrderedProductDTO] = interCaseClassDTO
+
+// outgoing events
```
Comment (nicolasfara) [inline production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala]: also here

**Review by nicolasfara — APPROVED** [review]

File: production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+final case class ProductionPlanReadyDTO(productionPlan: ProductionPlanDTO)
+final case class ProductionPlanDTO(productsToProduce: List[ProductToProduceDTO])
```
Comment (giacomocavalieri) [inline production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala:26]: You do not need this intermediate passage for the dto, simply 
```scala 
final case class ProductionPlanReadyDTO(productionPlan: List[ProductToProduceDTO])
```
could suffice; @nicolasfara which one do you prefer?

File: production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import dev.atedeg.mdm.productionplanning.dto.QuantityDTO.given
+import dev.atedeg.mdm.products.dto.ProductDTO
+import dev.atedeg.mdm.products.dto.ProductDTO.given
+import dev.atedeg.mdm.utils.serialization.DTO
+import dev.atedeg.mdm.utils.serialization.DTOGenerators.*
+import dev.atedeg.mdm.utils.serialization.DTOOps.*
+
+// incoming events
+final case class NewOrderReceivedDTO(order: OrderDTO)
+final case class OrderDTO(orderID: String, requiredBy: String, orderedProducts: List[OrderedProductDTO])
```
Comment (giacomocavalieri) [inline production-planning/src/main/scala/dev/atedeg/mdm/productionplanning/dto/DTOs.scala:17]: You do not need this intermediate passage, simply 
```scala 
final case class NewOrderReceivedDTO(orderID: String, requiredBy: String, orderedProducts: List[OrderedProductDTO])
```
could suffice; I'm not sure which style is better though

**Review by giacomocavalieri — APPROVED** [review]

**Pull request merged**